servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -207,26 +243,22 @@ public void stop() throws IOException {
   }
 
   /**
-   * Completes a file checkpoint in ufs. Called via RPC.
+   * Persists a file in UFS.
    *
-   * @param workerId the worker id completing the ufs checkpoint
-   * @param fileId the file id associated with the ufs checkpoint
+   * @param fileId the file id
    * @param length the length of the file
-   * @param checkpointPath the ufs path of the file
    * @return true on success
    * @throws SuspectedFileSizeException
    * @throws BlockInfoException
    * @throws FileDoesNotExistException
    */
-  public boolean completeFileCheckpoint(long workerId, long fileId, long length,
-      TachyonURI checkpointPath)
-          throws SuspectedFileSizeException, BlockInfoException, FileDoesNotExistException {
+  public boolean persistFile(long fileId, long length) throws SuspectedFileSizeException,
+      BlockInfoException, FileDoesNotExistException {
     synchronized (mInodeTree) {
       long opTimeMs = System.currentTimeMillis();
-      LOG.info(FormatUtils.parametersToString(workerId, fileId, length, checkpointPath));
-      if (completeFileCheckpointInternal(workerId, fileId, length, checkpointPath, opTimeMs)) {
+      if (persistFileInternal(fileId, length, opTimeMs)) {
```

Comment:
But that method name is already in use. This method used to be called addCheckpoint and Bin suggested to rename it to persistFile. What it really does is that it is the callback from the BlockDataManager that the final has been moved to its final location.

The current flow of file creation / completion / persistence seems rather convoluted. The client closes the file, which informs the worker to move the finle to its final destination and independently informs the master that the file can be completed. After the worker moves the file to its final destionation, it informs the master the file has been "finalized" (currently called persist file). Do you have any suggestion as to what the ``persistFile`` function should be called instead?